src/sentry/tasks/statistical_detectors.py

```diff
@@ -92,14 +126,12 @@ def run_detection() -> None:
     performance_projects_count = 0
     profiling_projects_count = 0
 
-    for project in RangeQuerySetWrapper(
-        Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
-        step=100,
-    ):
+    for project, project_settings in all_projects_with_settings():
         if project.flags.has_transactions and (
             features.has(
                 "organizations:performance-statistical-detectors-ema", project.organization
             )
+            and project_settings["duration_regression_detection_enabled"]
```

Comment:
Should this have a more specific name? As we can more duration regression type issues in the future?

Reply:
updated with `transaction_duration_regression_detection_enabled` 👍